File: DsaProblemsInJava/src/main/java/org/example/Jan012025/Jan162025.java
package org.example.Jan012025;

public class Jan162025 {
    public int xorAllNums(int[] nums1, int[] nums2) {
        int result = 0;

        // Check if the length of nums1 and nums2 are even or odd
        if (nums1.length % 2 == 0 && nums2.length % 2 == 0) {
            return 0;
        }

        // Case where nums1 is even and nums2 is odd
        if (nums1.length % 2 == 0 && nums2.length % 2 != 0) {
            for (int elem : nums1) {
                result ^= elem;
            }
        }

        // Case where nums1 is odd and nums2 is even
        if (nums1.length % 2 != 0 && nums2.length % 2 == 0) {
            for (int elem : nums2) {
                result ^= elem;
            }
        }

        // Case where both nums1 and nums2 are odd
        if (nums1.length % 2 != 0 && nums2.length % 2 != 0) {
            for (int elem : nums1) {
                result ^= elem;
            }
            for (int elem : nums2) {
                result ^= elem;
            }
        }

        return result;
    }
}

File: DsaProblemsInJava/src/main/java/org/example/Jan012025/PythonCodes/Jan162025.py
def xorAllNums(self, nums1: List[int], nums2: List[int]) -> int:
    result = 0
    if(len(nums1)%2==0 and len(nums2)%2==0):
        return 0
    if(len(nums1)%2==0 and len(nums2)%2!=0):
        for elem in nums1:
            result = result ^ elem
    if(len(nums1)%2!=0 and len(nums2)%2==0):
        for elem in nums2:
            result = result ^ elem
    if(len(nums1)%2!=0 and len(nums2)%2!=0):
        for elem in nums1:
            result = result ^ elem
        for elem in nums2:
            result = result ^ elem

    return result
